Sources/SQLKit/Query/SQLQueryString.swift

```diff
@@ -1,65 +1,130 @@
     mutating public func appendLiteral(_ literal: String) {
-        fragments.append(.literal(literal))
+        self.fragments.append(SQLRaw(literal))
     }
     
+    /// Adds an interpolated string of raw SQL. Despite the use of the term "literal" dictated by the interpolation
+    /// protocol, this produces `SQLRaw` content, _not_ SQL string literals.
     mutating public func appendInterpolation(_ literal: String) {
-        fragments.append(.literal(literal))
+        self.fragments.append(SQLRaw(literal))
     }
-
+    
+    /// Embed an `Encodable` value as a binding in the SQL query.
     mutating public func appendInterpolation(bind value: Encodable) {
-        fragments.append(.value(value))
+        self.fragments.append(SQLBind(value))
     }
 
-    /// Binds multiple values in a comma separated list.
-    /// Commonly used with the `IN` operator.
+    /// Embed multiple `Encodable` values as bindings in the SQL query, separating the bind placeholders with commas.
+    /// Most commonly useful when working with the `IN` operator.
     mutating public func appendInterpolation(binds values: [Encodable]) {
-        fragments.append(.values(values))
+        self.fragments.append(SQLList(values.map(SQLBind.init)))
+    }
+    
+    /// Embed an integer as a literal value, as if via `SQLLiteral.numeric()`
+    /// Use this preferentially to ensure values are appropriately represented in the database's dialect.
+    mutating public func appendInterpolation<I: BinaryInteger>(literal: I) {
+        self.fragments.append(SQLLiteral.numeric("\(literal)"))
+    }
+
+    /// Embed a `Bool` as a literal value, as if via `SQLLiteral.boolean()`
+    mutating public func appendInterpolation(_ value: Bool) {
+        self.fragments.append(SQLLiteral.boolean(value))
+    }
+
+    /// Embed a `String` as a literal value, as if via `SQLLiteral.string()`
+    /// Use this preferentially to ensure string values are appropriately represented in the database's dialect.
+    mutating public func appendInterpolation(literal: String) {
+        self.fragments.append(SQLLiteral.string(literal))
+    }
+
+    /// Embed an array of `Strings` as a list of literal values, using the `joiner` to separate them.
+    ///
+    /// Example:
+    ///
+    ///     "SELECT \(literals: "a", "b", "c", "d", joinedBy: "||") FROM nowhere"
+    ///
+    /// Rendered by the SQLite dialect:
+    ///
+    ///     SELECT 'a'||'b'||'c'||'d' FROM nowhere
+    mutating public func appendInterpolation(literals: [String], joinedBy joiner: String) {
+        self.fragments.append(SQLList(literals.map(SQLLiteral.string(_:)), separator: SQLRaw(joiner)))
+    }
+
+    /// Embed a `String` as an SQL identifier, as if with `SQLIdentifier`
+    /// Use this preferentially to ensure table names, column names, and other non-keyword identifiers are appropriately
+    /// represented in the database's dialect.
+    mutating public func appendInterpolation(ident: String) {
```

Comment:
Fantastic PR. The only thing that sticks out to me is that `ident` isn't exactly the most obvious label here. Does Xcode autocomplete this? If so, we can probably keep it, otherwise, maybe we should use a different label.